ProjectName/src/screens/PaymentDetails/PaymentDetailsForm/components/customAmountInput/CustomAmountInput.tsx

```diff
@@ -0,0 +1,53 @@
+import React, {FC} from 'react';
+import {Controller} from 'react-hook-form';
+import {View, Text, TextInput, Image} from 'react-native';
+
+import {DECIMAL_REGEX} from './custom-amount-input.constants';
+import {styles} from './custom-amount-input.styles';
+import {CustomMessageInputTypes} from '../customMessageInput/custom-message-input.types';
+
+const CustomAmountInput: FC<CustomMessageInputTypes> = ({
+  control,
+  name,
+  title,
+  placeholder,
+}) => {
+  return (
+    <View>
+      <Text style={styles.title}>{title}</Text>
+      <Controller
+        control={control}
+        name={name}
+        render={({field: {value, onChange, onBlur}}) => (
+          <View style={styles.container}>
+            <TextInput
+              keyboardType="decimal-pad"
+              value={value}
+              onChangeText={text => {
+                if (DECIMAL_REGEX.test(text)) {
+                  const formattedValue = text;
+                  onChange(formattedValue);
+                }
+              }}
+              onBlur={onBlur}
+              placeholder={placeholder}
+              maxLength={12}
+              style={styles.textAmountInput}
+            />
+            <View style={styles.currencyIconContainer}>
+              <View style={styles.currencyIcon}>
+                <Image
+                  source={require('../../../../../assets/pictures/FlagofEurope.png')}
```

Comment:
Please create `index.ts` file inside `pictures` directory and reexport pictures from there, so you could then `import` images as regular module. (you could check how it's implemented in our project - you might need to add some *.d.ts file to satisfy TS, though).